services/autoscaling/src/simcore_service_autoscaling/background_task.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+import contextlib
+import datetime
+import logging
+from typing import Awaitable, Callable
+
+from servicelib.logging_utils import log_context
+
+logger = logging.getLogger(__name__)
+
+
+async def _repeated_scheduled_task(
+    task: Callable[..., Awaitable[None]],
+    *,
+    interval: datetime.timedelta,
+    task_name: str,
+    **kwargs,
+):
+    while await asyncio.sleep(interval.total_seconds(), result=True):
```

Comment:
Q: wouldn;t you like it to start right away? This pattern is wait, run, wait, run ...  while I think if run, wait, run, wait. ... is more convenient specially when you restart a service because e.g. it failed, and you want to "get to work"  right away.

Reply:
well in this precise case. I think this is irrelevant.
But you are right, that maybe I should change that to some kind of ```do ... while```

Reply:
replaced by tenacity